### PR TITLE
Add canary badge to data plane

### PR DESCRIPTION
**Pull request by ferhoyos**
### Describe the change

PR changes:
- Add canary revision badge to dataplane. It is necessary to know which dataplane belongs to each revision.

image

- Show mesh link in tooltips only when the user is not in the mesh page (except the IstioStatus at masthead). It does not make sense to show 'More info at mesh page' when you are already on the mesh page...

- Convert TargetDataplane to React hook
- Convert IstioStatus to React hook
- Clean code (IstioStatusInline) and lint issues

### CODE REVIEW

File: mesh/config/cytoscape/cytoscape.go [hunk 1]
```diff
@@ -40,6 +40,7 @@ type NodeData struct {
 	InfraData      interface{} `json:"infraData,omitempty"`      // infraType-dependent data
 	IsAmbient      bool        `json:"isAmbient,omitempty"`      // true if configured for ambient
 	IsBox          string      `json:"isBox,omitempty"`          // set for NodeTypeBox, current values: [ 'cluster', 'dataplanes', 'namespace' ]
+	IsCanary       bool        `json:"isCanary,omitempty"`       // true if the data plane is canary
```
Comment (jshaughn) [inline mesh/config/cytoscape/cytoscape.go:43]: I guess another way we could have done this is to introduce another InfraType, like `MeshInfraType.DATAPLANECANARY`.  Although, I'm not sure that's better, this should be fine.

File: mesh/generator/generator.go [hunk 2]
```diff
@@ -108,18 +105,18 @@ func BuildMeshMap(ctx context.Context, o mesh.Options, gi *mesh.AppenderGlobalIn
 		// convert istio status slice into map
 		healthData := make(map[string]string)
 		for _, data := range istioStatus {
-			// istiod health depends on istiod and istio-pod status (both starts with "istiod" prefix)
-			if strings.HasPrefix(data.Name, "istiod") {
+			// istiod health depends on istiod and istio-pod status (both starts with the istiod name of the control plane)
+			if strings.HasPrefix(data.Name, cp.IstiodName) {
```
Comment (jshaughn) [inline mesh/generator/generator.go:109]: +1, this is good.

**Review by jshaughn — APPROVED** [review]

**Review by jshaughn — APPROVED** [review]
This all looks good, @ferhoyos , gracias!  I had only one comment but I don't think it would really be a simplifier.

File: mesh/config/cytoscape/cytoscape.go (hunk 1 above)
Comment (ferhoyos) [reply mesh/config/cytoscape/cytoscape.go:43]: I think that for the moment it is better to have IsCanary as an attribute. Currently both dataplanes are exactly the same except the badge, and adding new infratype would add more complexity in the frontend (add MeshInfraType.DATAPLANECANARY to all switches we have DATAPLANE). In a new mesh page iteration, if we decide to display different information for canary data plane, I am ok to add new infraType.

Comment (jshaughn) [reply mesh/config/cytoscape/cytoscape.go:43]: No, I think 'isCanary` is fine, it was just a thought...

**Pull request merged**